cvmfs/network/download.cc

```diff
@@ -1888,15 +1916,68 @@ Failures DownloadManager::Fetch(JobInfo *info) {
     //          info->wait_at[0], info->wait_at[1]);
     pipe_jobs_->Write<JobInfo*>(info);
 
+    bool is_running = true;
     do {
```

Comment:
This loop should go in its own method

Reply:
true.. all of it needs some clean up if we really found a solution that is at least equal and better (at least in some situations) then the old version